### ISSUE TITLE
deposit: sharing of uploads

### ISSUE THREAD

**Issue by lnielsen** [zenodo/zenodo#151]
Allow an upload to be shared among multiple users. Complementary to allowing a curator of a community to edit a deposit.

**Comment by peterdesmet** [comment]
Yes! Would also very much like this.

**Comment by guptavaibhav18197** [comment]
@lnielsen Sir
I would like to add this enhancement.
Can you please give a description like what/how will those multiple users be able to edit the metadata? What are edit rights of the curator of that community? and
Where can I find the related code?

**Comment by moble** [comment]
It would be particularly nice if this could be tied to github organizations, which are accessible via the github oauth API.  So, for example, if I want my whole community or just one record to be shared by my github organization, Zenodo could check the github organization memberships of any user attempting to alter a record.

Alternatively, this could be tied to Zenodo communities.  Once [community curation](https://github.com/zenodo/zenodo/issues/35) is enabled, perhaps a deposit could just be given to the community, so that anyone who can curate the community can alter any deposit within that community.

**Comment by peterdesmet** [comment]
GitHub orgs or Zenodo communities would be nice, but for my use cases these would include too many people for a single deposit. It also differs to much deposit vs deposit.

But having one person starting a deposit and then being able to grant view/write/admin permissions for each co-author would be excellent.

I would implement it via ORCID: link your ORCID to Zenodo account => you get view/write/admin access (depending on how they were set) to the deposits in which you are co-author (with your ORCID).

**Comment by maurice-schleussinger** [comment]
As I don't know what  the "Triage:Accepted" label signifies: is this planned for a release in the foreseeable future?

**Comment by slint** [comment]
@maurice-schleussinger The `Triage: Accepted` label means we will address the issue, but we're not yet sure on which Sprint/Roadmap this will be done

**Comment by zaneselvans** [comment]
So until the shared management of archives is integrated into Zenodo, how do y'all recommend dealing with collectively published & updated materials? Should we create an organizational account using an email alias and share the login credentials? Or if you're highly confident that this feature will get implemented in the near term... should we just create the archives as individuals, and assume that we'll be able to add other maintainers in the future? What do folks typically do, and are there any known pitfalls?  I searched around the Zenodo Help and FAQs and didn't find any suggestions... but maybe I missed them?

**Comment by lnielsen** [comment]
This will be implemented as part of the InvenioRDM project with the first release in summer 2020. Zenodo is likely to run on top of InvenioRDM by late autumn 2020 with this feature included.

Until then, shared accounts is, unfortunately, the only way.

**Comment by RoyMudie** [comment]
Realise this was meant to get started during COVID so anticipate delays - but is the InverioRDM functionality implmeneted?

Shared accounts isn't a solution if you only find out about this aspect AFTER you've published stuff... :\

**Comment by turian** [comment]
I need this too

**Comment by bricksdont** [comment]
Any update on this? Can I now find and use this feature on Zenodo?

**Comment by skchronicles** [comment]
@lnielsen
Are there any updates to this specific feature?

This would be extremely useful, as most projects work in a team.

**Comment by lnielsen** [comment]
In InvenioRDM you can now share a link to view, preview or edit a record (see screenshot below, and you can test it out on https://inveniordm.web.cern.ch):

<img width="905" alt="Screenshot 2022-07-25 at 09 15 54" src="https://user-images.githubusercontent.com/1698163/180719762-9d67b66d-6fb1-45d4-b5d3-3c4df45cf778.png">

Communities in InvenioRDM now also has teams with roles support (e.g. all curators can edit records in the community). See https://inveniordm.docs.cern.ch/releases/versions/version-v9.0.0/#members

The idea is still to also be able to individually share a single upload with multiple users as described in this issue. Most of the features needed for this is already implemented (e.g. a search for users to share an upload with, permissions and storing of permissions in the record). That said, the feature is not currently available on Zenodo and all focus is on making current Zenodo features available in InvenioRDM so we can migrate as fast as possible, hence it's not likely to start development until after we have migrated (unless some RDM partners jumps on it and implements it). We'll be communicating during Aug/Sep much more about the move to InvenioRDM, but brought ETA is about ~1 year.

**Comment by MLopez-Ibanez** [comment]
> It has been a bit quite ... can anyone give an update? Right now, we do share the login and password in plain text by email forth and back in order to kind of "collaborate" ... it feels slightly sub-optimal.

My understanding as an outsider is that all effort is focused right now towards moving to the new system and once the move happens, lots of new features will become available (like this one). If you wish to test the new system see: https://zenodo-rdm.web.cern.ch/

I think issues can be reported here: https://github.com/zenodo/zenodo-rdm

**Comment by samueljohn** [comment]
Thank you @lnielsen and @MLopez-Ibanez, fair enough. I understand that for the deposits that we have already published we continue to just share a login/password among team.

**Comment by ljgarcia** [comment]
Hi all, is this feature now available? If yes, where can I find some documentation? If not, is there a timeline? Thanks.

**Comment by uniqueg** [comment]
As much as I love Zenodo: It's been 9 years since this issue was opened. Sharing admin rights to maintain a record sustainibly is arguably an absolute core feature for any registry. I would bet that, currently, the vast majority of records are managed by single users and those who really care about data preservation are forced to share credentials - that's pretty much a nightmare either for the very idea of Zenodo - or for data security and integrity. Actually, in my opinion, uploaders should be very strongly encouraged (if not _forced_) to specify at least one co-owner for each upload/record.

In timelines of decades, people move on, die (unfortunately)... So please, can we at least have an update on how this is envisioned to work (or does it already work?!), with a timeline for integration?

Note also that share links are not a proper substitute for allowing co-owners/-admins.

Thanks a lot!

**Comment by uniqueg** [comment]
Actually, by creating Communities for your teams this _seems_ to work. One can add people to the community and give them relatively fine-grained roles (reader, curator, maintainer, owner). So, for a team-centric approach at least, this seems to do the trick. But need to test it a bit further.

It would be good to document this though, and promote/encourage co-ownership of data. And respond to this issue :pray:

**Comment by moble** [comment]
> Actually, by creating Communities for your teams this _seems_ to work.

Have you tried it?  When it says "curate records", I believe that just means accept or reject requests to add a record to the community — not to actually edit records in that community.

---

***EDIT***: My belief was evidently wrong.  See @uniqueg's comment below.  The only thing I would add is that I have a very old community, and community curation *did* work for me, unlike uniqueg's report.

**Comment by moble** [comment]
The InvenioRDM docs say that the metadata key `owned-by` [can have cardinality (1-n)](https://inveniordm.docs.cern.ch/reference/metadata/#parent) — meaning at least one, but possibly many owners.  So *maybe* there's a way to do it through the API.  On the other hand, it looks like the metadata the Zenodo API returns just has `'owner': <integer>`, so only one appears to be allowed.

**Comment by uniqueg** [comment]
Okay, reporting back with our experiences...

### Our use case

So, we are a research group where we wanted each group member to manage their Zenodo submissions themselves, while still giving the PI and the lab manager owner permissions for each record.

***This particular use case is indeed addressed perfectly through Communities!***

### What we did

We followed the steps described below:

1. I have created a [new community](https://zenodo.org/communities/new).
2. After giving a name and filling in some "Settings", I headed over to tab "Members".
3. I invited our PI in _Owner_ role. The available roles are:
   ![image](https://github.com/zenodo/zenodo/assets/10855418/2440d29c-90fc-4056-8b95-0603e7db2dd9)
4. I have asked the group members to link their records to the Community through the "Communities" card in the sidebar on the right of each record.
6. Afer approving the record in the group, both my PI and myself were able to edit the records, create new versions etc., just as if we had submitted the record ourselves:
   ![image](https://github.com/zenodo/zenodo/assets/10855418/890b9e44-b5ae-4660-8d01-00ac16b0fd54)

As a nice side effect, all our records will nicely be lumped together in one space! :+1: 

### Points to consider

- Importantly, community owners were able to edit records _even if the original submitters who linked the records to the community were not themselves part of the community_!
- On the other hand, I am an owner of another, much older community - and I am unable to edit the records in _that_ community for some reason! I am currently not sure if this is because the old community somehow does not benefit this feature (the admin panel looks exactly the same though), or whether the people linking the records can choose to give community owners the rights to edit the linked records (I didn't ask them yet).
- The community approach works great for teams where the people who should have privileged access are always the same. If your use case requires you to to co-own/co-manage records with a different combination of people each time, the incurred overhead will surely be annoying. In that case, link sharing (which I didn't try) might be a better option:
   ![image](https://github.com/zenodo/zenodo/assets/10855418/21178151-4575-42b4-ba2b-254785c75b68)

**Comment by joshmoore** [comment]
Thanks for this report, @uniqueg!

> Importantly, community owners were able to edit records even if the original submitters who linked the records to the community were not themselves part of the community!

For me this could conceivably be considered a secvuln. ⚠️ I as a user do **NOT** expect that a community member can edit my entries *without even changing the version number!*

**Comment by n8upham** [comment]
Hi all -- are these any updates on whether this 'community curation' tool has come to fruition on Zenodo?  I'd like to be able to add a community of curators to an existing Zenodo record, with the ability to specify different levels of permissions to the various members (e.g., edit vs publish).  Related to ongoing requests in #810 and #35.  Thanks!

**Comment by moble** [comment]
> are these any updates on whether this 'community curation' tool has come to fruition on Zenodo?

Yeah, it seems to work.  See @uniqueg's [comment](https://github.com/zenodo/zenodo/issues/151#issuecomment-1982138636) just above for a pretty good exploration of this.
